pkg/operator2/configobservation/apiserver/observe_cors.go

```diff
@@ -0,0 +1,60 @@
+package apiserver
+
+import (
+	"k8s.io/klog"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/openshift/cluster-authentication-operator/pkg/operator2/configobservation"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+var clusterDefaultCORSALlowedOrigins = []string{
+	`//127\.0\.0\.1(:|$)`,
+	`//localhost(:|$)`,
+}
+
+func ObserveAdditionalCORSAllowedOrigins(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	const corsAllowedOriginsPath = "corsAllowedOrigins"
+
+	listers := genericListers.(configobservation.Listers)
+	errs := []error{}
+	defaultConfig := map[string]interface{}{}
+	if err := unstructured.SetNestedStringSlice(defaultConfig, clusterDefaultCORSALlowedOrigins, corsAllowedOriginsPath); err != nil {
+		// this should not happen
+		return defaultConfig, append(errs, err)
+	}
+
+	// grab the current CORS origins to later check whether they were updated
+	currentCORSAllowedOrigins, _, err := unstructured.NestedStringSlice(existingConfig, corsAllowedOriginsPath)
+	if err != nil {
+		return defaultConfig, append(errs, err)
+	}
+	currentCORSSet := sets.NewString(currentCORSAllowedOrigins...)
+	currentCORSSet.Insert(clusterDefaultCORSALlowedOrigins...)
+
+	observedConfig := map[string]interface{}{}
+	apiServer, err := listers.APIServerLister().Get("cluster")
+	if errors.IsNotFound(err) {
+		klog.Warningf("apiserver.config.openshift.io/cluster: not found")
+		return defaultConfig, errs
+	}
+	if err != nil {
+		return existingConfig, errs
```

Comment:
The error should probably be appended, correct.

`existingConfig` is returned because returning `defaultConfig` might rewrite whatever we had even if `err` is just a transient network error, and we would have another rewrite once that is over. I'll add a comment.